core/src/com/unciv/ui/MultiplayerScreen.kt

```diff
@@ -431,6 +439,66 @@ class EditMultiplayerGameInfoScreen(game: GameInfo, gameName: String, backScreen
             }
         }
     }
+
+    /**
+     * Helper function to decrease indentation
+     * Turns the current playerCiv into an AI civ and uploads the game afterwards.
+     */
+    private fun giveUp(gameId: String, gameName: String, backScreen: MultiplayerScreen){
+        //Create a popup
+        val popup = Popup(this)
+        popup.addGoodSizedLabel("Working...").row()
+        popup.open()
+
+        thread {
+            try {
+                //download to work with newest game state
+                val gameInfo = OnlineMultiplayer().tryDownloadGame(gameId)
+                val playerCiv = gameInfo.currentPlayerCiv
+
+                //only give up if it's the users turn
+                //this ensures that no one can upload a newer game state while we try to give up
+                if (playerCiv.playerId == game.settings.userId){
+                    //call next turn first to make sure everything works like intended
+                    //this skips everything like trade request which will be handled by the AI on the following turn
```

Comment:
Shouldn't we change the owner to AI, and only then next turn? Currently, the civilization effectively 'skips' a turn off decision making and unit movement

Reply:
Yea switching them around didn't work first and I didn't know why. I realized that GameInfo.nextTurn() needed a tweak to allow an AI civ to call nextTurn() without skipping its own turn.
Added in newest commit